lib/Report/SessionHistory.php

```diff
@@ -361,6 +361,87 @@ public function getResults(SanitizerInterface $sanitizedParams)
                 $rows[] = $logRecord;
             }
 
+            return new ReportResult(
+                $metadata,
+                $rows,
+                count($rows),
+            );
+        } else {
+            $params = [
+                'fromDt' => $fromDt->format(DateFormatHelper::getSystemFormat()),
+                'toDt' => $toDt->format(DateFormatHelper::getSystemFormat()),
+            ];
+
+            $sql = 'SELECT
+             `session_history`.`sessionId`,
+             `session_history`.`startTime`,
+             `session_history`.`userId`,
+             `session_history`.`userAgent`,
+             `session_history`.`ipAddress`,
+             `user`.`userName`,
+             `usertype`.`userType`,
+             `session`.`session_expiration`
+                FROM `session_history`
+                    LEFT OUTER JOIN `user` ON `user`.`userId` = `session_history`.`userId`
+                    LEFT OUTER JOIN `usertype` ON `usertype`.`userTypeId` = `user`.`userTypeId`
+                    LEFT OUTER JOIN `session` ON `session`.`session_data` 
+                        LIKE CONCAT("sessionHistoryId|i:", session_history.sessionId, "%")
+             WHERE `session_history`.startTime BETWEEN :fromDt AND :toDt
+             ';
+
+            if ($sanitizedParams->getInt('userId') !== null) {
+                $sql .= ' AND `session_history`.`userId` = :userId';
+                $params['userId'] = $sanitizedParams->getInt('userId');
+            }
+
+            if ($sanitizedParams->getInt('sessionHistoryId') !== null) {
+                $sql .= ' AND `session_history`.`sessionId` = :sessionHistoryId';
+                $params['sessionHistoryId'] = $sanitizedParams->getInt('sessionHistoryId');
+            }
+
+            // Sorting?
+            $sortOrder = $this->gridRenderSort($sanitizedParams);
+
+            if (is_array($sortOrder)) {
+                $sql .= ' ORDER BY ' . implode(',', $sortOrder);
+            }
+
+            $rows = [];
+            foreach ($this->store->select($sql, $params) as $row) {
+                $sessionRecord = $this->logFactory->createEmpty()->hydrate($row, ['htmlStringProperties' => ['message']]);
+                $endTime = isset($row['session_expiration']) ? date("Y-m-d H:i:s", $row['session_expiration']) : null;
```

Comment:
There are some stray `"` here.

Also it would be better to use the `getSystemFormat` function in DateFormatHelper